api/models/datastore.go

```diff
@@ -59,8 +56,44 @@ type Datastore interface {
 	// ErrDatastoreEmptyRoutePath when routePath is empty. Returns ErrRoutesNotFound when no route exists.
 	RemoveRoute(ctx context.Context, appID, routePath string) error
 
-	// GetDatabase returns the underlying sqlx database implementation
-	GetDatabase() *sqlx.DB
```

Comment:
the extension stuff (and in particular https://github.com/fnproject/ext-auth) was using this. i'm no fan either, but it may be better to wage that battle another time (up to you)

Reply:
Hmm good point - I think I chose to keep this broken in the very short term. 

One option might be to return the root data source from the stack via this interface and then let the extensions up-cast to something that gives them the DB